FreeRTOS-Plus/Source/FreeRTOS-Plus-TCP/FreeRTOS_ARP.c

```diff
@@ -624,11 +651,15 @@ ARPPacket_t *pxARPPacket;
 		xARPHeader.usOperation;
 		xARPHeader.xTargetHardwareAddress;
 	*/
-	memcpy( ( void * ) pxARPPacket, ( void * ) xDefaultPartARPPacketHeader, sizeof( xDefaultPartARPPacketHeader ) );
-	memcpy( ( void * ) pxARPPacket->xEthernetHeader.xSourceAddress.ucBytes , ( void * ) ipLOCAL_MAC_ADDRESS, ( size_t ) ipMAC_ADDRESS_LENGTH_BYTES );
-	memcpy( ( void * ) pxARPPacket->xARPHeader.xSenderHardwareAddress.ucBytes, ( void * ) ipLOCAL_MAC_ADDRESS, ( size_t ) ipMAC_ADDRESS_LENGTH_BYTES );
 
-	memcpy( ( void* )pxARPPacket->xARPHeader.ucSenderProtocolAddress, ( void* )ipLOCAL_IP_ADDRESS_POINTER, sizeof( pxARPPacket->xARPHeader.ucSenderProtocolAddress ) );
+	/* Also, for rule 21.15 regarding using same pointer-to-x types for memcpy,
+	 * below is done intentionally here and thus the rule is relaxed         */
+	/* coverity[misra_c_2012_rule_21_15_violation] */
```

Comment:
I don't believe you need to suppress this here.  Note that lines 659/660/662 do not conform to the prototype for `memcpy()` – the second argument should be cast to `( const void * ).  Correct that and you should be able tyo delete lines 655-657.

```c
       void *memcpy(void *dest, const void *src, size_t n);
```